apps/remix-ide/src/app/udapp/run-tab.js

```diff
@@ -201,6 +201,31 @@ export class RunTab extends LibraryPlugin {
     this.renderDropdown(this.udappUI, this.fileManager, this.compilersArtefacts, this.config, this.editor, this.logCallback)
     this.renderRecorder(this.udappUI, this.fileManager, this.config, this.logCallback)
     this.renderRecorderCard()
+
+    this.on('manager', 'pluginDeactivated', profile => {
+      if (profile.kind === 'provider') this.blockchain.removeProvider(profile.name)
+    })
+    this.on('manager', 'pluginActivated', profile => {
```

Comment:
suggestion: `this.on('manager', 'pluginActivated', this.addProfileProvider.bind(this)` and then move the code to that function